static/app/views/lastKnownRouteContextProvider.tsx

```diff
@@ -0,0 +1,29 @@
+import {createContext, useContext} from 'react';
+
+import getRouteStringFromRoutes from 'sentry/utils/getRouteStringFromRoutes';
+import usePrevious from 'sentry/utils/usePrevious';
+import {useRoutes} from 'sentry/utils/useRoutes';
+
+interface Props {
+  children?: React.ReactNode;
+}
+
+export const LastKnownRouteContext = createContext<string>('');
+
+export function useLastKnownRoute() {
+  return useContext(LastKnownRouteContext);
+}
+
+// This provider tracks the last known route that the user has navigated to.
+// This is used to better group issues when we hit "route not found" errors.
+export default function LastKnownRouteContextProvider({children}: Props) {
+  const route = useRoutes();
+  const prevRoute = usePrevious(route);
+  const lastKnownRoute = getRouteStringFromRoutes(prevRoute);
```

Comment:
I wish we could just use the location state, but we call navigate in so many areas that it would be really annoying to update